src/System.Linq/tests/DistinctTests.cs

```diff
@@ -164,5 +136,57 @@ public void CustomEqualityComparer()
 
             Assert.Equal(expected, source.Distinct(new AnagramEqualityComparer()), new AnagramEqualityComparer());
         }
+
+        [Theory, MemberData("SequencesWithDuplicates")]
+        public void FindDistinctAndValidate<T>(T unusedArgumentToForceTypeInference, IEnumerable<T> original)
+        {
+            // Convert to list to avoid repeated enumerations of the enumerables.
+            var originalList = original.ToList();
+            var distinctList = originalList.Distinct().ToList();
+
+            // Ensure the result doesn't contain duplicates.
+            var hashSet = new HashSet<T>();
+            foreach (var i in distinctList)
+            {
+                Assert.False(hashSet.Contains(i), "Found a duplicate in the result of Distinct(): " + i);
+                hashSet.Add(i);
```

Comment:
Rather than assertin on Contains, you could assert that Add returns true, right?